### DISCUSSION TITLE
VS2022 Windows DLL

### DISCUSSION THREAD

**Discussion by edumarzolla** [elalish/manifold#874] [General]
Dear all,
I am trying to compile manifold as a DLL to be used in C#.
The objective is to create a GLB file after creating cylinders and boolean operation with these cylinders.

When I compile using the sintax below, the dll is compiled, but without export funcions (that is necessary to export the GLB file)
cmake . -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=OFF -DMANIFOLD_DEBUG=ON -DMANIFOLD_PAR=TBB -DMANIFOLD_EXPORT=OFF -A x64 -B build

When I try to compile using the sintax below, the dll is not compiled because it is missing the assimp lib
cmake . -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=OFF -DMANIFOLD_DEBUG=ON -DMANIFOLD_PAR=TBB -DMANIFOLD_EXPORT=ON -A x64 -B build
the difference is only DMANIFOLD_EXPORT, form OFF to ON.

Could you please tell me how to enable the DMANIFOLD_EXPORT=ON (and probably DBUILD_SHARED_LIBS=ON also)?

Thanks,

Eduardo

**Comment by pca006132** [comment]
You need to install assimp. Perhaps https://github.com/assimp/assimp/blob/master/Build.md#install-on-all-platforms-using-vcpkg can help. I am not familiar with c++ dependency management on windows.

**Comment by edumarzolla** [comment]
Dear all,

I made a small change in manifoldDeps.cmake... and now the assimp lib is
installed automatically by VS2022.
Now the VS2022 compiled the DLL with the following command:



Please let me know if this change can be implemented on the future
versions of your wonderful lib Manifold!
Thank you!






Em seg., 22 de jul. de 2024 às 09:50, pca006132 ***@***.***>
escreveu:

**Reply by pca006132** [reply]
Sorry, but I can't see your cmake changes. Feel free to open a PR and we can discuss how to integrate it. I think it is fine to add this to the windows build as windows users may have difficulty installing packages.

**Reply by elalish** [reply]
Indeed, please push this in the form of a PR!
